flocker/cli/functional/test_deploy_script.py

```diff
@@ -29,13 +44,88 @@ class FlockerDeployTests(TestCase):
     """
     @_require_installed
     def setUp(self):
-        pass
+        self.persistence_service = ConfigurationPersistenceService(
+            reactor, FilePath(self.mktemp()))
+        self.persistence_service.startService()
+        self.cluster_state_service = ClusterStateService()
+        self.cluster_state_service.startService()
+        self.addCleanup(self.cluster_state_service.stopService)
+        self.addCleanup(self.persistence_service.stopService)
+        app = ConfigurationAPIUserV1(self.persistence_service,
+                                     self.cluster_state_service).app
+        api_root = Resource()
+        api_root.putChild('v1', app.resource())
+        self.port = reactor.listenTCP(0, Site(api_root),
+                                      interface="127.0.0.1")
+        self.addCleanup(self.port.stopListening)
+        self.port_number = self.port.getHost().port
```

Comment:
How would I find the port it was listening on? I guess I coudl do the find_free_port() hack or whatever it is...